placeSpecificContent/hq/chatterboxes/chatterbox/buttons/privacyButton_server.js

```diff
@@ -0,0 +1,404 @@
+                    x: doorCurrentLocalPosition.x,
+                    y: DEFAULT_DOOR_CLOSED_LOCAL_POSITION.y,
+                    z: doorCurrentLocalPosition.z
+                };
+                Entities.editEntity(door, {
+                    localPosition: doorCurrentLocalPosition
+                });
+            // If door is very close to correct position, slam into exact position
+            } else {
+                if (Vec3.distance(doorCurrentLocalPosition, doorMovements.targetPosition) <= 
+                    MIN_DISTANCE_TO_TRANSITION_ENTITIES_M) {
+                    Entities.editEntity(door, {
+                        localPosition: DEFAULT_DOOR_OPEN_LOCAL_POSITION
+                    });
+                }
+                doorCurrentLocalPosition = doorMovements.targetPosition;
+            }
+            _this.getDoorMovementData(doorMovements, doorCurrentLocalPosition);
+        },
+
+        /* Determine whether the door needs + or - velocity along each necessary axis to get to the target position */
+        getDoorMovementData: function(doorMovements, doorCurrentLocalPosition) {
+            if (doorCurrentLocalPosition.x !== doorMovements.targetPosition.x) {
+                doorMovements.xVelocity = (doorCurrentLocalPosition.x > doorMovements.targetPosition.x) ? 
+                    -MOVEMENT_LOCAL_VELOCITY_M_PER_S : MOVEMENT_LOCAL_VELOCITY_M_PER_S;
+            }
+            if (doorCurrentLocalPosition.z !== doorMovements.targetPosition.z) {
+                doorMovements.zVelocity = (doorCurrentLocalPosition.z > doorMovements.targetPosition.z) ? 
+                    -MOVEMENT_LOCAL_VELOCITY_M_PER_S : MOVEMENT_LOCAL_VELOCITY_M_PER_S;
+            }
+        },
+
+        /* Move entities and if they are all in place, end the transition */
+        updateEntities: function( doorMovements, shadeTargetHeight) {
+            var doorCurrentLocalPosition = Entities.getEntityProperties(door, 'localPosition').localPosition;
+
+            _this.maybeMoveDoorOnXAxis(doorMovements, doorCurrentLocalPosition);
+            _this.maybeMoveDoorOnZAxis(doorMovements, doorCurrentLocalPosition);
+            _this.maybeChangeShadeHeight(shadeTargetHeight);
+
+            if (!doorMovements.xVelocity && !doorMovements.zVelocity && shadeHeightChange === 0) {
+                _this.playSound(DONE_SOUND, DONE_VOLUME, buttonPosition, false, false);
+                currentState = (currentState === "toClosed") ? "closed" : "open";
+                Entities.editEntity(button, {
+                    color: currentState === "closed" ? BUTTON_CLOSED_COLOR : BUTTON_OPEN_COLOR
+                });
+                if (updateInterval) {
+                    Script.clearInterval(updateInterval);
+                    updateInterval = null;
+                }
+            }
+        },
+
+        /* If the door is moving, check its position along the x axis to see if it is place. If it is, check if we need 
```

Comment:
I'm confused about why this might need to move along the Z axis. Can you please help me understand?

Reply:
Not all pods are aligned along the same axis but they will be aligned along either the x or z axis because of audio zones. The doors move in two directions, in/out of the doorway, and over to slide away from the doorway. They will move along both axes because of this but we have to determine which axis a door moves on for the sliding over movement and which axis the door moves on to fit into the doorway according to the axis that the chatterbox is aligned to. Then, we still have to determine which axis the moves on first as it depends on whether the door is opening or closing. To open, the door has to move out away from the doorway, then over to the side and to close these movements are reversed. So, I keep track of which axis the door moves along first and when that is finished, check to see if it still needs to move along the other axis.